added a default_params config check from config.yaml file and nokogiri search for param values in source doc

jeffreycwitt opened the pull request:
added a default_params config check from config.yaml file and nokogiri aided retrieval into source doc for validating schema version and document type values before going to default_params. 

Readme file was also updated to address issue #1

jeffreycwitt committed:
added a default_params config check from config.yaml file; also added nokogiri aided retrieval into source doc for validating schema version and document type values before going to default_params

jeffreycwitt commented:
@stenskjaer can you look over this pr before I merge.

stenskjaer commented:
The added documentation is nice.
It complains about the nokogiri dependency. I have nokogiri (1.6.8) installed, but it coughs up this warning when running lbp-print:
``` log
➜  lbp-print git:(develop) ✗ lbp-print
lbp-print
WARN: Unresolved specs during Gem::Specification.reset:
      nokogiri (~> 1.6)
WARN: Clearing out unresolved specs.
Please report a bug if this causes problems.
```
But worse is that after that it errors out with this message:
``` log
/Library/Ruby/Gems/2.0.0/gems/lbp-print-0.1.0/bin/lbp-print:8:in `require': cannot load such file -- pry (LoadError)
	from /Library/Ruby/Gems/2.0.0/gems/lbp-print-0.1.0/bin/lbp-print:8:in `<top (required)>'
	from /usr/bin/lbp-print:23:in `load'
	from /usr/bin/lbp-print:23:in `<main>'
```
I cannot find `pry` even though I have installed it...
When I build without the `pry` dependency, it works (but still with the nokogiri warning).

By the way, great to include custom defaults!

stenskjaer committed:
Check if xpath is not empty

Otherwise it always evaluates to True as an empty path returns `[]`,
which I guess is interpreted as True.

stenskjaer commented:
Well... On further testing I found that the xpath looking into the XML always evaluates to True even when it returns an empty list. 

So I added a check for that that I thought would push to this PR, but of course it doesn't work like that when I have not made it myself. So it created a new PR that makes that change. I guess you can just merge it after this one, if you want, depending on how it evolves. Otherwise just copy my change into this and reject my PR.

stenskjaer committed:
Check for nested parameters

The has_key? does not validate to True on my box when the params are
nested in the yaml.

stenskjaer committed:
Forgot the last config check on `schema`

stenskjaer committed:
To string conversion of type on ascription

Otherwise the type is not converted to string (on my system)

stenskjaer committed:
Update whitespace

Sorry about that... again...

jeffreycwitt committed:
Merge pull request #5 from stenskjaer/develop

Update evaluations

jeffreycwitt committed:
removed pry dependency; added an initial if condition on all parameters to see if there was a command line override

jeffreycwitt commented:
Ok, I think it's even closer now. I'm not sure why are getting the nokogiri warning, anything about 1.6 should work. I'll keep thinking about it.

stenskjaer commented:
Actually I was playing with changing the positional arguments to optional arguments. 
You can see my preliminary ideas here: https://github.com/stenskjaer/lbp-print/blob/feature/option-flags/bin/lbp-print

This would remove the requirement of the positional sequence of the arguments. Besides, with the positional arguments, you cannot leave out any intermediate arguments if your want to define the later ones can you?
So if I wanted to set the schema, I could not just write `lbp-print tex 69-l1q1 0.0.0` because that would result in `output_parent=0.0.0`, which is not intended.
With positional arguments, I would just do `lbp-print tex 69-l1q1 --schema=0.0.0` or similar. I think that's neat.

With the optional arguments we can also add descriptions to them for the `help` commands, aliases and what not.
